rust/src/lib.rs

```diff
@@ -134,6 +134,16 @@ pub extern "C" fn ror_treefile_get_add_file_fd(
     fd.as_raw_fd()
 }
 
+#[no_mangle]
+pub extern "C" fn ror_treefile_get_passwd_fd(tf: *mut Treefile) -> libc::c_int {
+    tf_from_raw(tf).externals.passwd.as_ref().map_or(-1, |fd| fd.as_raw_fd())
```

Comment:
Should we always seek back to 0 here in case we're called multiple times?

Reply:
Pushed a new commit to do this consistently :arrow_down: